docs/language-server/DocumentEventSyncPublisherSubscriberDesign.md

```diff
@@ -0,0 +1,114 @@
+# Document event sync publisher subscriber
+
+## Overview
+
+This document contains information about the implementation of a notification system inside the language server using the 
+PubSub architecture. Goal is to allow communication between language server and extensions synchronously.
+
+Following diagram shows how this has been implemented. 
+
+![alt text](../images/language-server/001-DocumentEventPubSub-class-diagram.png)
+
+- Event Subscribers and Event Publishers get loaded using JavaSPI and get subscribed to publishers when a document 
+opens or changes.
+- Event Publishers and Subscribers have an Event kind and a subscriber can only subscribe to a single publisher at once.
+If we need a subscriber to subscribe for two publishers, then we have to create two different subscribers and subscribe
```

Comment:
Don't show the link between publisher and subscriber. Publisher should link to Event Kind. Also, Subscriber should link to Event Kind.

Reply:
For example, 

If we need to consume events from multiple EventKinds, then we have to create two different subscribers and subscribe to those EventKinds individually.